code-challenges.js

```diff
@@ -1,63 +1,131 @@
+    if( number % 3 === 0 ){
+        return `${number} is divisible by three`
+        }else{
+            return `${number} is not divisible by three`
+        }
+    }
+    //console log the function 
+     console.log(div3(num1))
+     console.log(div3(num2))
+     console.log(div3(num3))
 
 
 // --------------------2) Create a function that takes in the randomNouns variable and returns an array with all the words capitalized.
 
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
-// Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
-
-
-
-
+//  Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
+    
+//First you will declare a function that takes in an argument
+    const nounCap = (array) => {
+        //create a local variable that maps through the array
+        let eachNoun = array.map(value =>{
+            //capitalize the first index of each word
+            //then connect it to the rest of the word with substring method
+            return value[0].toUpperCase() + value.substring(1)
+            })
+            return eachNoun.join(" ")
+            
+         
+        
+    }
+        console.log(nounCap(randomNouns))
+    
 
 // --------------------3) Create a function that takes in the mixedDataArray array and returns an array with ONLY NUMBERS sorted from least to greatest.
 
 var mixedDataArray = [true, 8, "hello", 90, -8, null, 0, 46, 59, 107, "hey!"]
-// Expected output: [-8, 0, 8, 46, 59, 90, 107]
-
+//  Expected output: [-8, 0, 8, 46, 59, 90, 107]
 
+// declare a function that takes in an argument 
+  const onlySortNumbs = (array) =>{
+      //sort through each filter value to determine if the value is a string. you will use the type of 
+      return array.filter(value => 
+        typeof value === "number")
+        // sort through numbers with .sort() from least to greatest
+        .sort(function(a,b){return a-b})
+      
+    }
+console.log(onlySortNumbs(mixedDataArray))
 
 
 
 // --------------------4) Create a function that takes in a string and logs the index of the first vowel.
 
 var vowelTester1 = "learn"
-// Expected output: 1
+// // Expected output: 1
 var vowelTester2 = "throw"
-// Expected output: 3
-
-
-
-
+// // Expected output: 3
+ 
+ 
+//  //declare a function that takes in an argument
+ const indexFirstVowel = (string) => {
+    //make  for loop to loop through all the letters
+    for(let i = 0; i < string.length; i++){
+        //make a variable  with all the vowels
+        let vowels = "aeiou"
+        //make a variable that tells the character based on the index
+        let char = string.charAt(i)
+        // make an if statement that uses indexOf() to tell yu the virst instance of a vowel 
+        if ( vowels.indexOf(char.toLowerCase()) != -1 ){
```

Comment:
if you make this line `if ( vowels.indexOf(string[i].toLowerCase()) != -1 ){` it will work just the same :) 

but great job on the logic here. small refactors would make this code cleaner but it absolutely runs and gets the job done!